### PR TITLE
fix(core): fix a11y issue for tab component

### PR DESCRIPTION
**Pull request by manu-kr**
## Related Issue.
Closes SAP/fundamental-ngx/ https://github.com/SAP/fundamental-ngx/issues/6704

## Description

This PR change contains fix the accessibility issue those were reported by GB adoption team.
https://product-jira.ariba.com/browse/SS-27241
##### During Implementation
1. Visual Testing:
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [x] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [x] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [x] API boundary values
- [x] different combinations of components - free style
- [x] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [x] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application
- [NA] update `README.md`
- [NA] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [x] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [x] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [x] API boundary values
- [x] different combinations of components - free style
- [x] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/tabs/examples/index.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import { TabSelectionExampleComponent } from './tab-selection-example/tab-select
 import { TabsExampleComponent } from './tabs-example/tabs-example-component';
 import { TabCollapsibleExampleComponent } from './tab-collapsible-example/tab-collapsible-example.component';
 import { TabCollapsibleOverflowExampleComponent } from './tab-collapsible-overflow-example/tab-collapsible-overflow-example.component';
-
+import { TabFormElementExampleComponent } from './tab-form-element-example/tab-form-element-example.component';
 export * from './tab-collapsible-overflow-example/tab-collapsible-overflow-example.component';
```
Comment (Lokanathan-k) [inline apps/docs/src/app/core/component-docs/tabs/examples/index.ts:19]: leave a space of line between export and import statement

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/** This is just for fixing the space issue in tab content, not recomdended **/
+.fd-content-button {
+    margin-right: 16px !important;
+    margin-top: 16px !important;
+}
```
Comment (Lokanathan-k) [inline apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss:5]: is this required ?

Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss]: removed.

File: apps/docs/src/app/core/component-docs/tabs/examples/index.ts (hunk 1 above)
Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/tabs/examples/index.ts]: modified.

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Component } from '@angular/core';
+
+@Component({
+    selector: 'fd-tab-form-element-example',
+    templateUrl: './tab-form-element-example.component.html',
+    styleUrls: ['./tab-form-element-example.component.scss']
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.ts:6]: build is still failing. The .scss file was deleted but is still being referenced here.

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss (hunk 2 above)
Comment (platon-rov) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss]: Please use REM.

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.ts (hunk 3 above)
Comment (platon-rov) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.ts:6]:
```
Error: Entry module not found: Error: Can't resolve '/Users/madrov/coding/sap/fundamental-ngx/apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss' in '/Users/madrov/coding/sap/fundamental-ngx'
```

File: libs/core/src/lib/tabs/tab-link/tab-link.directive.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ import { AbstractFdNgxClass } from '@fundamental-ngx/core/utils';
     selector: '[fd-tab-link]',
     host: {
         role: 'tab',
-        tabindex: '0'
+        tabindex: '-1'
```
Comment (platon-rov) [inline libs/core/src/lib/tabs/tab-link/tab-link.directive.ts:19]: After this change tabbing though tabs is not possible anymore.

File: libs/core/src/lib/tabs/tab-list.component.html [hunk 5]
```diff
@@ -13,7 +13,7 @@
                  [attr.aria-selected]="tab.active">
                 <a
                     fd-tab-link
-                    [attr.tabindex]="0"
+                    [attr.tabindex]= "tab.active ? '0':'-1'"
```
Comment (platon-rov) [inline libs/core/src/lib/tabs/tab-list.component.html:16]: Tabbing not possible anymore.

**Review by platon-rov — CHANGES_REQUESTED** [review]

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss (hunk 2 above)
Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.scss]: removed

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.ts (hunk 3 above)
Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-form-element-example/tab-form-element-example.component.ts]: modified.

File: libs/core/src/lib/tabs/tab-link/tab-link.directive.ts (hunk 4 above)
Comment (manu-kr) [reply libs/core/src/lib/tabs/tab-link/tab-link.directive.ts:19]: Yes that's the requirement. here we have to tab using left and right arrow key. Please refer below link for more details.
https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html

File: libs/core/src/lib/tabs/tab-list.component.html (hunk 5 above)
Comment (manu-kr) [reply libs/core/src/lib/tabs/tab-list.component.html:16]: Yes that's the requirement. here we have to tab using left and right arrow key. Please refer below link for more details.
https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html

**Review by Lokanathan-k — APPROVED** [review]
LGTM

**Review by platon-rov — CHANGES_REQUESTED** [review]
For me VoiceOver on Mac doesn't recognize amount of tabs properly.

image

**Comment by manu-kr** [comment]
> For me VoiceOver on Mac doesn't recognize amount of tabs properly.
> 
> image

This is working correctly in both chrome and JAWS screen reader.

On Mac doesn't recognising nested elements with role="tab" inside tablist. If i move role=tab to fd-tab-item directive the screenreader will read it twice  once as a tab and once as a link. and it's not acceptable. Please let me on your thoughts on this. 
Screenshot 2021-09-24 at 4 48 09 PM

**Comment by platon-rov** [comment]
> This is working correctly in both chrome and JAWS screen reader.
> 
> On Mac doesn't recognising nested elements with role="tab" inside tablist. If i move role=tab to fd-tab-item directive the screenreader will read it twice once as a tab and once as a link. and it's not acceptable. Please let me on your thoughts on this.
> Screenshot 2021-09-24 at 4 48 09 PM

In SAPUI5 we have hidden elements with the appropriate ARIA information which are referenced then by `aria-labelledby`\`aria-describedby` like below:

```
<span aria-labelledby="hidden-elem">I need additional information and I cannot be properly managed by the ARIA attributes, custom text should be read!</span>

<span aria-hidden="true" id="hidden-elm">Additional info which will be read</span>
```

In that case we will normalise the way how is the ARIA information is read by the different screen-readers. But it takes some additional `@Input()`s to cover i18n with the values for that hidden elements. It adds additional work on our side and on the customer side, so I'm not fully sure about it. Maybe it's totally OK to support JAWS + Chrome Screenreader. 

@droshev @InnaAtanasova maybe some thoughts?

**Review by Betrozov — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Comment by platon-rov** [comment]
@manu-kr I've approved, we've discussed issue I mentioned, later with the a11y team will be agreed on how to cover all screen-readers, it's okay to have jaws supported for now.

**Review by puru-hk — APPROVED** [review]

**Pull request merged**